src/components/Toast/Toast.tsx

```diff
@@ -1,97 +1,99 @@
+import { NOOP } from "../../utils/function-utils";
 import "./Toast.scss";
 
-const defaultIconMap = {
-  [TOAST_TYPES.NORMAL]: Info,
-  [TOAST_TYPES.POSITIVE]: Check,
-  [TOAST_TYPES.NEGATIVE]: Alert
-};
-
-const getIcon = (type, icon) => {
-  /* icon may be node a may be a string */
-  if (icon && typeof icon === "object") {
-    return icon;
-  }
-  return icon || defaultIconMap[type] ? (
-    <Icon
-      iconType={icon ? Icon.type.ICON_FONT : Icon.type.SVG}
-      clickable={false}
-      icon={icon || defaultIconMap[type]}
-      iconSize="24px"
-      ignoreFocusStyle
-    />
-  ) : null;
-};
+interface ToastProps extends VibeComponentProps {
+  actions?: ToastAction[];
+  /** If true, Toast is open (visible) */
+  open?: boolean;
+  type?: ToastType;
+  /** Possible to override the default icon */
+  icon?: string | React.FC<IconSubComponentProps> | null;
+  /** If true, won't show the icon */
+  hideIcon?: boolean;
+  /** The action to display */
+  action?: JSX.Element;
+  /** If false, won't show the close button */
+  closeable?: boolean;
+  onClose?: () => void;
+  /** The number of milliseconds to wait before
+   * automatically closing the Toast
+   * (0 or null cancels this behaviour) */
+  autoHideDuration?: number;
+  children?: ReactElement | ReactElement[] | string;
+}
 
-const Toast = ({
-  open,
-  autoHideDuration,
-  type,
+const Toast: FC<ToastProps> & { types?: typeof ToastType; actionTypes?: typeof ToastActionType } = ({
+  open = false,
+  autoHideDuration = null,
+  type = ToastType.NORMAL,
   icon,
-  hideIcon,
+  hideIcon = false,
   action: deprecatedAction,
   actions,
   children,
-  closeable,
-  onClose,
+  closeable = true,
+  onClose = NOOP,
   className
 }) => {
   const toastLinks = useMemo(() => {
     return actions
       ? actions
-          .filter(action => action.type === TOAST_ACTION_TYPES.LINK)
+          .filter(action => action.type === ToastActionType.LINK)
           .map(({ type: _type, ...otherProps }) => <ToastLink key={otherProps.href} {...otherProps} />)
       : null;
   }, [actions]);
-  const toastButtons = useMemo(() => {
+
+  const toastButtons: JSX.Element[] | null = useMemo(() => {
     return actions
       ? actions
-          .filter(action => action.type === TOAST_ACTION_TYPES.BUTTON)
+          .filter(action => action.type === ToastActionType.BUTTON)
           .map(({ type: _type, content, ...otherProps }, index) => (
             <ToastButton key={`alert-button-${index}`} {...otherProps}>
               {content}
             </ToastButton>
           ))
       : null;
   }, [actions]);
+
   const classNames = useMemo(
     () => cx("monday-style-toast", `monday-style-toast--type-${type}`, className),
     [type, className]
   );
+
   const handleClose = useCallback(() => {
     if (onClose) {
       onClose();
     }
   }, [onClose]);
+
   /* Timer */
-  const timerAutoHide = useRef();
+  const timerAutoHide = useRef<NodeJS.Timeout>();
```

Comment:
why nodeJS timeout

Reply:
Cause it's the return type of `setTimeout`: 
`export function setTimeout<[]>(     callback: () => void,     ms?: number): NodeJS.Timeout`